packages/core/src/state-management/repository.ts

```diff
@@ -265,6 +284,36 @@ export class Repository {
     return state$
   }
 
+  async resumeRunningStatesFromAnchorRequestStore(): Promise<void> {
+    const currentTimestamp = Date.now()
+    let gt: StreamID | undefined = undefined
+    let batch = new Array<AnchorRequestStoreListResult>()
+    do {
+      batch = await this.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
+      batch.map((listResult) => {
+        this.resumeQ.add(async () => {
+          const stream$ = await this.load(listResult.key, { sync: SyncOptions.PREFER_CACHE })
+          if ([AnchorStatus.FAILED, AnchorStatus.ANCHORED].includes(stream$.value.anchorStatus)) {
```

Comment:
I am a bit conflicted about this. Could you tell please how it relates to recovery mechanism in `fromStreamStateStore`?

Reply:
I'm not sure how it relates to the recovery mechanism from the `fromStreamStateStore`, @ukstv . 

To me, this particular `if` block is just a sanity check, as there should be no anchor requests in storage for streams with anchor status `FAILED` or `ANCHORED` in the first place. I'm happy to remove this check here or replace it with a logger warning.